### PR TITLE
🐛 Fix TestServerSideApplyWithDefaulting flakes

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR is an attempt to get rid of TestServerSideApplyWithDefaulting flakes.

The assumption is that the test case "no-op_if_neither_is_defaulted" is moving too quickly from creating the webhook and calling NewServerSidePatchHelper, which expect the web up to be already registered/recognized by the API server.

The fix introduces as wait loop after creating the web hook to make sure the webhook is up and running before moving on.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/10902

/area clusterclass

### CODE REVIEW

File: internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go [hunk 1]
```diff
@@ -679,22 +679,37 @@ func TestServerSideApplyWithDefaulting(t *testing.T) {
...
+			// NOTE: we are doing this no matter of defaultOriginal to make sure that the web hook is up and running
+			// before calling NewServerSidePatchHelper down below.
+			g.Eventually(ctx, func(g Gomega) {
+				// create a new object
+				kct2 := kct.DeepCopy()
+				kct2.Name = fmt.Sprintf("%s-2", kct.Name)
+				g.Expect(env.Create(ctx, kct2)).To(Succeed())
+				defer func() {
+					_ = env.Delete(ctx, kct2)
+				}()
```
Comment (chrischdi) [inline internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go:692]: We could use server-side dry-run to not need to delete it :-)

File: internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go [hunk 2]
```diff
@@ -679,22 +679,37 @@ func TestServerSideApplyWithDefaulting(t *testing.T) {
...
+
+				// check if it was defaulted
+				g.Expect(env.Get(ctx, client.ObjectKeyFromObject(kct2), kct2)).To(Succeed())
```
Comment (sbueringer) [inline internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go:695]:
```suggestion
```
kct2 is updated by the create call

Comment (sbueringer) [reply internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go:695]: Together with Christian's comment this should work
```go
			g.Eventually(ctx, func(g Gomega) {
				// create a new object
				kct2 := kct.DeepCopy()
				kct2.Name = fmt.Sprintf("%s-2", kct.Name)
				g.Expect(env.Create(ctx, kct2, client.DryRunAll)).To(Succeed())
				g.Expect(kct2.Spec.Template.Spec.Users).To(BeComparableTo([]bootstrapv1.User{{Name: "default-user"}}))
			}, 5*time.Second).Should(Succeed())
```

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by chrischdi** [comment]
/lgtm cancel

see above comments :-)

**Comment by sbueringer** [comment]
Thx! and nice idea with the dry-run @chrischdi !

/lgtm
/approve

**Pull request merged**